addons/project/models/project_project.py

```diff
@@ -24,6 +24,7 @@ class Project(models.Model):
         'mail.thread',
         'mail.activity.mixin',
         'mail.tracking.duration.mixin',
+        'analytic.plan.fields.mixin',
```

Comment:
The distribution is on the project?
Why? It really doesn't make sense to me.

Imagine you have a Project for your client Deco Addict.
You have one task to repair some stuff, one task to build something for them, and one task to check up on them to see if they need new stuff.
Each task will be done by a different department even though it is in the same project.

Reply:
I understand that could be strange but in fact, the purpose of this task is to facilitate the using of the analytic and to avoid having analytic distribution everywhere and we choose the project because we think it would be the best candidate to have the default analytic distribution for the different documents (SO, PO, MO, BoM).

And so, 
- in Sales app, we add (again) a project field on SOs and we can set by default an analytic distribution on their lines according to the config of the project linked,
- in Purchase, same idea than SOs, we add the project on PO and we can set by default an analytic distribution on their lines according to the config of the project linked,
- in MRP, we replace the analytic distribution field in MO and BoM by project. By doing that, we can easier track the costs/revenues to display them inside the project profitability by using the project field. 🙂

According to your example, I would not set anything except the `account_id` to be able to timesheets, in the second task I could set an SOL to do the research and in that SOL, I alter the analytic distribution for instance. 🤷🏼‍♂️ 
(it is just my opinion, maybe we could do it in another way)

Reply:
I wonder if that would not fall into the distribution models then?

Reply:
Good question. 🤷🏼‍♂️

Reply:
If it's the case, this whole PR doesn't make sense anymore and can be replaced by something simpler.
After talking about it with other devs, we think it makes sense, but there are no PO working ATM.

Reply:
Okay, I did not know what `distribution models` model does. If we just use the project to define the default value for analytic distribution in different models then okay you right but here it is not really the case, we remove the analytic distribution in MO and BoM to have project field. By doing that, we can link those records to a project (before we have to check if the analytic account of the project is in the analytic distribution to know if "perhaps" the MO is linked to the project) and so for a construction company, everything will be available in the project.

Moreover, in that PR, we use your mixin to be able to add the configuration as default value for the analytic distribution in SOLs, POLs.

Also, the fields added by your `analytic.plan.fields.mixin` mixin are not required, the only field we need in our features is `account_id` field to be able to timesheets inside the project and so the configuration could be almost and the user can just set the project to a MO, BoM, SO, PO to be able to have it inside his project thanks to the embedded actions displayed in a top bar.

![image](https://github.com/user-attachments/assets/c105c699-9eda-4982-b626-102e75eaa7a1)

cc @deneuvillem @auon-odoo